openslides_backend/action/actions/user/forget_password.py

```diff
@@ -0,0 +1,136 @@
+from ...mixins.send_email_mixin import EmailMixin, EmailSettings
+from ...util.default_schema import DefaultSchema
+from ...util.register import register_action
+from ...util.typing import ActionData
+
+PW_FORGET_EMAIL_TEMPLATE = """You receive this email, because you have requested a new password for your OpenSlides-account.
+
+Please open the following link and choose a new password:
+/login/forget-password-confirm?user_id={user_id}&token={token}
+
+For completeness your username: {username}"""
+PW_FORGET_EMAIL_SUBJECT = "Reset your OpenSlides password"
+
+
+class format_dict(defaultdict):
+    def __missing__(self, key: str) -> str:
+        return f"'{key}'"
+
+
+@register_action("user.forget_password")
+class UserForgetPassword(EmailMixin, UpdateAction):
+    """
+    Action to send forget password mail(s).
+    """
+
+    model = User()
+    schema = DefaultSchema(User()).get_default_schema(
+        title="user forget password schema",
+        additional_required_fields={"email": {"type": "string"}},
+    )
+    skip_archived_meeting_check = True
+
+    def get_updated_instances(self, action_data: ActionData) -> ActionData:
+        for instance in action_data:
+            email = instance.pop("email")
+
+            # check emailif valid email adress
+            if not self.check_email(EmailSettings.default_from_email):
+                raise ActionException(
+                    f"email '{EmailSettings.default_from_email}' is not a valid sender email address."
+                )
+            if not self.check_email(email):
+                raise ActionException(f"'{email}' is not a valid email adress.")
+
+            # search for users with email
+            filter_ = FilterOperator("email", "=", email)
+            results = self.datastore.filter(
+                self.model.collection, filter_, ["id", "username"]
+            )
+
+            # try to send the mail.
+            try:
+                with EmailMixin.get_mail_connection() as mail_client:
+                    for user in results.values():
+                        if self.sendmail(
+                            mail_client,
+                            email,
+                            self.get_email_subjext(),
+                            self.get_email_body(
+                                user["id"],
+                                self.get_token(user["id"], email),
+                                user["username"],
+                            ),
+                        ):
+                            yield {"id": user["id"], "last_email_send": round(time())}
+            except SMTPAuthenticationError as e:
+                raise ActionException(f"SMTPAuthenticationError: {str(e)}")
+            except SMTPSenderRefused as e:
+                raise ActionException(f"SMTPSenderRefused: {str(e)}")
+            except SSLCertVerificationError as e:
+                raise ActionException(f"SSLCertVerificationError: {str(e)}")
+            except Exception as e:
```

Comment:
Can this whole exception handling be extracted into a function in the email mixin? I don't think it's good to handle all exception in every function calling the mixin.

Reply:
Simplify the exception code here. Don't leak the exception to the user. In the general case (here). log the error and show a generic error message.